apps/studio/components/interfaces/Settings/Vault/Secrets/EditSecretModal.tsx

```diff
@@ -61,9 +72,14 @@ const EditSecretModal = ({ selectedSecret, onClose }: EditSecretModalProps) => {
     }
     payload.secret = values.secret
 
-    if (!isEmpty(payload)) {
+    if (!isEmpty(payload) && selectedSecret) {
       setSubmitting(true)
-      const res = await vault.updateSecret(selectedSecret.id, payload)
+      const res = await updateSecret({
+        projectRef: project?.ref!,
```

Comment:
`.?!` maybe instead of this we could have a simple if check that returns early if the project is missing (which should really never happen)

Reply:
yup! good catch